lang/semgrep-grammars/src/semgrep-c-sharp/grammar.js

```diff
@@ -40,7 +45,16 @@ module.exports = grammar(standard_grammar, {
     identifier: ($, previous) => {
       return choice(
         previous,
-        token(/\$[A-Z_][A-Z_0-9]*/)
+        $._semgrep_metavariable
+      );
+    },
+
+    _semgrep_metavariable: $ => token(/\$[A-Z_][A-Z_0-9]*/),
+
+    parameter: ($, previous) => {
+      return choice(
+        previous,
```

Comment:
Yes, aliases are completely ignored by ocaml-tree-sitter. Initially I tried to preserve them but it turned out to be complicated and fragile/broken. Field is also something that's ignored.

When `previous` is a `choice()`, we can extend it rather than creating a new one. It's not the case here, `previous` here being a `seq()`:
```js
    parameter: $ => seq(
      repeat($.attribute_list),
      optional($.parameter_modifier),
      optional(field('type', $._type)),
      field('name', $.identifier),
      optional($.equals_value_clause)
    ),
```
If we could call it `parameter_orig` or something like that, it would be great. The [tree-sitter-typescript grammar](https://github.com/tree-sitter/tree-sitter-typescript/blob/master/common/define-grammar.js) is written as a massive extension of tree-sitter-javascript, so maybe there's an example in there that would get us close. Otherwise, I think it's worth [asking the tree-sitter folks](https://github.com/tree-sitter/tree-sitter/discussions) directly.